kinisi/parser.py

```diff
@@ -347,3 +361,77 @@ def get_indices(
         drift_indices = sc.Variable(dims=['atom'], values=drift_indices)
 
         return indices, drift_indices
+
+
+def _get_molecules(structure: "ase.atoms.Atoms" or "pymatgen.core.structure.Structure"
+                   or "MDAnalysis.universe.Universe", coords: sc.Variable, indices: sc.Variable,
+                   masses: sc.Variable) -> Tuple[np.ndarray, np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+    """
+    Determine framework and non-framework indices for an :py:mod:`ase` or :py:mod:`pymatgen` or :py:mod:`MDAnalysis` compatible file when specie_indices are provided and contain multiple molecules. Warning: This function changes the structure without changing the object.
+
+    :param structure: Initial structure.
+    :param coords: fractional coordinates for all atoms.
+    :param indices: indices for the atoms in the molecules in the trajectory used in the calculation 
+    of the diffusion.
+    :param masses: Masses associated with indices in indices.
+    :param framework_indices: Indices of framework to be used in drift correction. If set to None will return all indices that are not in indices.
+
+
+    :return: Tuple containing: Tuple containing: fractional coordinates for centers and framework atoms
+    and Tuple containing: indices for centers used in the calculation 
+    of the diffusion and indices of framework atoms.
+    """
+    drift_indices = []
+    try:
+        indices.values = indices.values - 1
+    except:
+        raise ValueError('Molecules must be of same length')
+
+    n_molecules = len(indices.dims)
```

Comment:
I think that this is a bug and should read:
```suggestion
    n_molecules = indices.shape[0]
```
To reproduce the current implementation.